plugins/content_distribution/providers/tvinci/lib/feed/TvinciDistributionFeedHelper.php

```diff
@@ -57,34 +57,88 @@ private function createArgumentsForXSLT()
 		$arguments = array(
 			"distributionProfileId" => $this->distributionProfile->id,
 			"playManifestPrefix" => $prefix);
-		if($this->distributionProfile->ipadnewPpvModule)
-			$arguments["ipadnewPpvModule"] = $this->distributionProfile->ipadnewPpvModule;
-		if($this->distributionProfile->ipadnewFileName)
-			$arguments["ipadnewTypeName"] = $this->distributionProfile->ipadnewFileName;
-		if($this->distributionProfile->ismPpvModule)
-			$arguments["ismPpvModule"] = $this->distributionProfile->ismPpvModule;
-		if($this->distributionProfile->ismFileName)
-			$arguments["ismTypeName"] = $this->distributionProfile->ismFileName;
-		if($this->distributionProfile->iphonenewPpvModule)
-			$arguments["iphonenewPpvModule"] = $this->distributionProfile->iphonenewPpvModule;
-		if($this->distributionProfile->iphonenewFileName)
-			$arguments["iphonenewTypeName"] = $this->distributionProfile->iphonenewFileName;
-		if($this->distributionProfile->mbrPpvModule)
-			$arguments["mbrPpvModule"] = $this->distributionProfile->mbrPpvModule;
-		if($this->distributionProfile->mbrFileName)
-			$arguments["mbrTypeName"] = $this->distributionProfile->mbrFileName;
-		if($this->distributionProfile->dashPpvModule)
-			$arguments["dashPpvModule"] = $this->distributionProfile->dashPpvModule;
-		if($this->distributionProfile->dashFileName)
-			$arguments["dashTypeName"] = $this->distributionProfile->dashFileName;
-		if($this->distributionProfile->widevinePpvModule)
-			$arguments["widevinePpvModule"] = $this->distributionProfile->widevinePpvModule;
-		if($this->distributionProfile->widevineFileName)
-			$arguments["widevineTypeName"] = $this->distributionProfile->widevineFileName;
-		if($this->distributionProfile->widevineMbrPpvModule)
-			$arguments["widevineMbrPpvModule"] = $this->distributionProfile->widevineMbrPpvModule;
-		if($this->distributionProfile->widevineMbrFileName)
-			$arguments["widevineMbrTypeName"] = $this->distributionProfile->widevineMbrFileName;
+
+		$tagsDoc = new DOMDocument();
+		$tagsConfiguration = $tagsDoc->createElement('tagsconfiguration');
+		foreach($this->distributionProfile->tags as $tag)
+		{
+			/**
+			 * @var KalturaTvinciDistributionTag $tag
+			 */
+			$tagXML = $tagsDoc->createElement('tag');
+
+			$tagname = $tagsDoc->createElement('tagname');
```

Comment:
create a function "append tag element" or something like that for all these similar calls